packages/hub/node-tests/git/writer-remote-test.ts

```diff
@@ -19,342 +26,355 @@
-//   let env: todo, writers: todo, repo: Repository, tempRepoPath, tempRemoteRepoPath: string;
-
-//   this.timeout(10000);
-
-//   beforeEach(async function() {
-//     let root = await temp.mkdir('cardstack-server-test');
-
-//     let { repo: remoteRepo } = await makeRepo(root);
-
-//     let remote = await Remote.create(remoteRepo, 'origin', 'http://root:password@localhost:8838/git/repo');
-//     await remote.push('refs/heads/master', 'refs/heads/master', { force: true });
-
-//     let factory = new JSONAPIFactory();
-
-//     tempRepoPath = await mkdir('cardstack-temp-test-repo');
-//     tempRemoteRepoPath = await mkdir('cardstack-temp-test-remote-repo');
-
-//     repo = await Repository.clone('http://root:password@localhost:8838/git/repo', tempRemoteRepoPath);
-
-//     let dataSource = factory.addResource('data-sources').withAttributes({
-//       'source-type': '@cardstack/git',
-//       params: {
-//         remote: {
-//           url: 'http://root:password@localhost:8838/git/repo',
-//           cacheDir: tempRepoPath,
-//         },
-//       },
-//     });
-
-//     factory
-//       .addResource('content-types', 'events')
-//       .withRelated('fields', [
-//         factory.addResource('fields', 'title').withAttributes({ fieldType: '@cardstack/core-types::string' }),
-//         factory.addResource('fields', 'published-date').withAttributes({ fieldType: '@cardstack/core-types::string' }),
-//       ])
-//       .withRelated('data-source', dataSource);
-
-//     env = await createDefaultEnvironment(`${__dirname}/..`, factory.getModels());
-//     writers = env.lookup('hub:writers');
-//   });
-
-//   afterEach(async function() {
-//     await temp.cleanup();
-//     await destroyDefaultEnvironment(env);
-//     service.clearCache();
-//   });
-
-//   describe('create', function() {
-//     it('allows you to create a record in an empty git repo', async function() {
-//       let { data: record } = await writers.create(env.session, 'events', {
-//         data: {
-//           type: 'events',
-//           attributes: {
-//             title: 'Fresh Event',
-//             'published-date': '2018-09-01',
-//           },
-//         },
-//       });
-//       await repo.fetchAll();
-//       let saved = await inRepo(tempRemoteRepoPath).getJSONContents(
-//         'origin/master',
-//         `contents/events/${record.id}.json`
-//       );
-//       expect(saved).to.deep.equal({
-//         attributes: {
-//           title: 'Fresh Event',
-//           'published-date': '2018-09-01',
-//         },
-//       });
-//     });
-//   });
-// });
+import { makeRepo, inRepo } from './support';
+import { dir as mkTmpDir, DirectoryResult } from 'tmp-promise';
+
+const repoRealm = `${myOrigin}/api/realms/test-git-repo`;
+
+async function resetRemote() {
+  let tmpDir = await mkTmpDir({ unsafeCleanup: true });
+  let root = tmpDir.path;
+
+  let tempRepo = await makeRepo(root, {
+    'contents/events/event-1.json': JSON.stringify(
```

Comment:
in general I highly suggest using `CardDocument` to create test cards--this way you can be sure the card json is legit--like it has cs-fields, schema definitions, and adoption specification, all of which you are missing in this test data. that was its original purpose (it's branched out since then). There are lots of examples in hub/node-tests/card-service-tests.ts. the simplest form of it will create the field schema for you:
```ts
let doc = cardDocument().withAutoAttributes({ title: 'This is a test event', 'published-date': '2018-09-25' });
let jsonapi = doc.jsonapi;
```

Reply:
Yeah definitely meaning to update to that, I have a _lot_ of cleanup to do here from the old tests